sources/Common/DYNLoadIncreaseResult.h

```diff
@@ -29,80 +29,40 @@ namespace DYNAlgorithms {
 class LoadIncreaseResult {
  public:
   /**
-   * @brief default constructor
+   * @brief constructor
    */
-  LoadIncreaseResult();
+  explicit LoadIncreaseResult(const SimulationResult& preScenariosLoadIncreaseResult, const size_t nbScenarios);
 
-  /**
-   * @brief default destructor
-   */
-  ~LoadIncreaseResult();
-
-  /**
-   * @brief get the load level of the load increase
-   * @return load level of the load increase
-   */
-  double getLoadLevel() const;
-
-  /**
-   * @brief set the load level of the load increase
-   * @param loadLevel value to apply
-   */
-  void setLoadLevel(double loadLevel);
-
-  /**
-   * @brief get the status of the load increase
-   * @return status of the load increase
-   */
-  status_t getStatus() const;
-
-  /**
-   * @brief set the status of the load increase
-   * @param status value to apply
-   */
-  void setStatus(status_t status);
-
-  /**
-   * @brief specify the number of scenarios
-   * @param nbScenarios number of scenarios
-   */
-  void resize(size_t nbScenarios);
+  const SimulationResult& getPreScenariosLoadIncreaseResult() const {
+    return preScenariosLoadIncreaseResult_;
+  }
 
-  /**
-   * @brief getter of the number of scenarios
-   * @return the number of scenarios
-   */
-  size_t size() const {return results_.size();}
+  void setPreScenariosLoadIncreaseResult(const SimulationResult& preScenariosLoadIncreaseResult) {
+    preScenariosLoadIncreaseResult_ = preScenariosLoadIncreaseResult;
+  }
 
   /**
    * @brief result getter
    * @param idx id of the result
    * @return result object with index idx associated to this load increase
    */
-  SimulationResult& getResult(size_t idx);
+  SimulationResult& getScenarioResult(size_t idx);
 
   /**
    * @brief getter for the results
    * @return results associated to this load increase
    */
-  const std::vector<SimulationResult>& getResults() const;
-
-  /**
-   * @brief begin iterator on simulation results
-   * @return begin iterator on the first result
-   */
-  std::vector<SimulationResult>::const_iterator begin() const;
+  const std::vector<SimulationResult>& getScenariosResults() const {
+    return scenariosResults_;
+  }
 
-  /**
-   * @brief end iterator on simulation results
-   * @return end iterator on the first result
-   */
-  std::vector<SimulationResult>::const_iterator end() const;
+  void clearScenariosResults() {
+    scenariosResults_.clear();
+  }
 
  private:
-  std::vector<SimulationResult> results_;  ///< list of scenario results
-  status_t status_;  ///< status of the load increase
-  double loadLevel_;  ///< value of the variation of the load increase
+  SimulationResult preScenariosLoadIncreaseResult_;
```

Comment:
please name this result_

Reply:
I find the names a bit unclear. Do you think it could be a good idea to rename the LoadIncreaseResult class SessionResult.

Then, the SessionResult contains these attributes :

- SimulationResult loadIncreaseResult_
- std::vector<SimulationResult> scenariosResults_

Reply:
Or we could rename the SimulationResult class ScenarioResult and then rename the LoadIncreaseResult class SimulationResult.

Reply:
a load increase is a simulation, as well as the scenarios, so no need to change the class names. The preScenarios used in the current name will be unclear for most users, as well as "Session" which is not a keyword of the margin calculation. Maybe just rename this attribute loadIncreaseResult_ then?